proto/frequenz/api/weather/weather.proto

```diff
@@ -23,8 +23,9 @@ import "google/protobuf/timestamp.proto";
 //     Weather forecasts are inherently uncertain and actual conditions may
 //     vary. Use the data responsibly.
 service WeatherForecastService {
-  // Streams the weather forecast features for a geo location.
-  rpc SubscribeWeatherForecast(ForecastRequest)
+  // Streams live weather forecast features for a geo location as they become
+  // available. Initially, the most recent forecast will be streamed.
+  rpc StreamLiveWeatherForecast (ForecastRequest)
```

Comment:
I'm not sure I agree with using Publish/Subscribe in this context. Please see the explanations below. I think we'd be mixing up two concepts here if we'd start using Publish/Subscribe in APIs that are not centred around the concept of messaging platforms.

Stream
The term "Stream" is more generic and could be thought of as a continuous flow of data. When you say a service will "stream" data, you generally mean that it will keep sending data as long as the client is connected or until the data ends. In your use case, StreamLiveWeatherForecast effectively communicates that it's a live, continuous feed of weather forecast data.

Publish/Subscribe
"Publish/Subscribe," on the other hand, is a messaging pattern often used in more complex event-driven architectures. In a Pub/Sub system, publishers send messages to a channel without having knowledge of what subscribers there may be. Similarly, subscribers express interest in one or more channels and only receive messages

Reply:
The problem with the term `Stream` is that it is ambiguous. When a client invokes `server. stream_data()`, it is not clear enough that the client wants to publish data or subscribe data. The idea with the Publish/Subscriber terminology is that Publish methods accept a stream as input, and Subscribe methods return a stream. This is clearer that using `Stream` as a verb.

Note: this was originally discussed here: https://github.com/frequenz-floss/frequenz-api-microgrid/issues/84

Reply:
Also, in gRPC, if a method is returning a stream (i.e., clients _subscriber_ to data), then in the return type, the `stream` keword is present. So our methods could look like `rpc PublishX(params) returns (stream X)`. What I mean to say is that we still specify it in the API specs that we are dealing with data streams.

Reply:
> The problem with the term Stream is that it is an ambiguous verb

This is a standard term when designing APIs and widely understood. I don't think we should reinvent the wheel here.

Reply:
Agree that this is a standard term in APIs. However, in gRPC, it is hardly used as a stand-alone verb. It is always used [with qualifiers](https://grpc.io/docs/what-is-grpc/core-concepts/#service-definition) (client-side, server-side, biredtional).
So even in gRPC specs, the word `stream`, alone, is considered ambiguous.

Reply:
Whats most commonly used are these three types of streams in the context of APIs:

1. Client-to-Server Streaming
2. Server-to-Client Streaming
3. Bidirectional Streaming

Reply:
Exactly. And just the verb `Stream`, when used alone, does not convey which type of stream it is.

Reply:
> So even in gRPC specs, the word stream, alone, is considered ambiguous.

Agreed. I'm just not agreeing on using Publish/Subscribe in this context.

Reply:
Just adding an example here to demonstrate the ambiguity with `Stream`:
```proto
rpc StreamClientData(stream ClientData) returns (Empty); // client-side streaming
rpc StreamServerData(Params) returns (stream ServerData); // server-side streaming
```

Reply:
> Just adding an example here to demonstrate the ambiguity with Stream

To me the above it completely clear.  Where is the problem?

Reply:
What it demonstrates is that the same verb `Stream` is used in two different methods that behave completely differently. This is the ambiguity that we were talking about.

Reply:
> What it demonstrates is that the same verb Stream is used in two different methods that behave completely differently.

Well, how about the usage of "GET" or any other verb. Any verb carries that problem.

Reply:
> Well, how about the usage of "GET" or any other verb. Any verb carries that problem.

This actually highlights another problem. We generally name our non-streaming methods as `GetX` , `SetX`, `UpdateX`, etc. These verbs are activities that the client does (e.g., the client _gets_ a resource).
`StreamX` breaks the pattern. This is something that the server does, and not the client.

Reply:
Since it's clear that we need to discuss this further, I will resolve this discussion, to unblock this PR.